tests/manage/test_dynamic_pvc_accessmodes_with_reclaim_policies.py

```diff
@@ -0,0 +1,375 @@
+        set to Delete/Retain
+        Steps:
+        1. Create Storage Class with reclaimPolicy: Delete/Retain
+        2. Create PVC with 'accessModes: ReadWriteOnce'
+        3. Create two pods using same PVC
+        4. Run IO on first pod
+        5. Verify that second pod is not in Running state
+        6. Delete first pod
+        7. Wait for second pod to be in Running state
+        8. Verify usage of volume in second pod is matching with usage in
+           first pod
+        9. Run IO on second pod
+        10. Delete second pod
+        11. Delete PVC
+        12. Verify PV associated with deleted PVC is also deleted/released
+        """
+        # Create Storage Class with reclaimPolicy: Delete
+        logger.info(
+            f"Creating Storage Class with reclaimPolicy: {reclaim_policy}"
+        )
+        if interface_type == constants.CEPHBLOCKPOOL:
+            secret_name = self.rbd_secret_obj.name
+            interface_name = self.cbp_obj.name
+
+        elif interface_type == constants.CEPHFILESYSTEM:
+            secret_name = self.cephfs_secret_obj.name
+            interface_name = helpers.get_cephfs_data_pool_name()
+
+        sc_obj = helpers.create_storage_class(
+            interface_type=interface_type,
+            interface_name=interface_name,
+            secret_name=secret_name,
+            reclaim_policy=reclaim_policy
+        )
+
+        # Create PVC with 'accessModes' 'ReadWriteOnce'
+        logger.info('Creating PVC with accessModes: ReadWriteOnce')
+        pvc_data = templating.load_yaml_to_dict(constants.CSI_PVC_YAML)
+        pvc_data['metadata']['name'] = helpers.create_unique_resource_name(
+            'test', 'pvc'
+        )
+        pvc_data['metadata']['namespace'] = self.namespace
+        pvc_data['spec']['storageClassName'] = sc_obj.name
+        pvc_data['spec']['accessModes'] = ['ReadWriteOnce']
+        pvc_obj = PVC(**pvc_data)
+        pvc_obj.create()
+
+        # Create two pods using same PVC
+        logger.info(f"Creating two pods using same PVC {pvc_obj.name}")
+
+        # Create first pod
+        if interface_type == constants.CEPHBLOCKPOOL:
+            pod_data = templating.load_yaml_to_dict(constants.CSI_RBD_POD_YAML)
+        elif interface_type == constants.CEPHFILESYSTEM:
+            pod_data = templating.load_yaml_to_dict(
+                constants.CSI_CEPHFS_POD_YAML
+            )
+
+        pod_data['metadata']['name'] = helpers.create_unique_resource_name(
+            'test', 'pod'
+        )
+        pod_data['metadata']['namespace'] = self.namespace
+        pod_data['spec']['volumes'][0]['persistentVolumeClaim']['claimName'] = (
+            pvc_obj.name
+        )
+        pod_obj = helpers.create_pod(**pod_data)
+
+        node_pod1 = pod_obj.get()['spec']['nodeName']
+
+        # Create second pod
+        # Try creating pod until it is on a different node than first pod
+        for retry in range(1, 6):
```

Comment:
The range specified is hard coded. For example, can we use the values defined in ocs_basic_install.yml for worker/master nodes?....

Reply:
removed the for loop, second pod will be created on different node now